server/index.js

```diff
@@ -19,6 +19,51 @@ const jsonMiddleWare = express.json();
 app.use(staticMiddleware);
 app.use(jsonMiddleWare);
 
+app.get('/api/accounts/:accountId', (req, res, next) => {
+  const accountId = req.params.accountId;
+  if (!accountId) {
+    throw new ClientError(400, 'username must be provided');
+  }
+
+  const total = `select count(*) as "reviews"
+  from "reviews"
+  where "accountId" = $1
+  group by "accountId"`;
+
+  const sql = `
+    select "stateCode",
+    count(*) as "visits"
+    from "accounts"
+    join "reviews" using ("accountId")
+    join "parksCache" using ("parkCode")
+    where "accountId" = $1
+    group by "stateCode"
+    order by "visits" desc`;
+
+  const params = [accountId];
+  db.query(sql, params)
+    .then(result => {
+      const visits = result.rows;
+      if (!visits) {
+        res.status(404).json({
+          error: 'Cannot find reviews/states for account'
+        });
+      }
+      db.query(total, params)
+        .then(response => {
+          if (!response) {
+            res.status(404).json({
+              error: 'Cannot find reviews for account'
+            });
+          }
```

Comment:
If a `db.query()` becomes fulfilled, the `Result` object will always be present, event if its `.rows` is empty.